File: app/src/main/java/com/reclycer/repertoire/MainActivity.java
package com.reclycer.repertoire;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.Menu;
import android.view.MenuItem;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.list) RecyclerView rv;
    private MyAdapter ma = new MyAdapter();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);
        ma.refreshContact(this);
        rv.setLayoutManager(new LinearLayoutManager(this));
        rv.setAdapter(ma);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);

           return true;
    }

    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {

            case R.id.action_add:
                Intent intent = new Intent(this, AjoutContact.class);
                startActivityForResult(intent, 1);

                return true;
            case R.id.Delete:
                Intent intent1 = new Intent(this, DeleteActivity.class);
                startActivityForResult(intent1, 2);
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode) {
            case (1): {
                if (resultCode == Activity.RESULT_OK) {
                    ma.refreshContact(this);
                }
                break;
            }
            case(2): {
                if (resultCode == Activity.RESULT_OK) {
                    ma.refreshContact(this);
                    }
                }
                break;
        }
    }
 }

File: app/src/main/java/com/reclycer/repertoire/MainActivity.kt
package com.reclycer.repertoire

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.util.Log
import android.view.Menu
import android.view.MenuItem

import com.reclycer.repertoire.data.ContactService
import com.reclycer.repertoire.data.DataManager
import io.reactivex.SingleObserver
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val myAdapter = MyAdapter()
    lateinit var datamanager: DataManager
    lateinit var databasemanager: DatabaseManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        contact_list.layoutManager = LinearLayoutManager(this)
        contact_list.adapter = myAdapter

        databasemanager = DatabaseManager(this)
        displayUpdatedContent()

        datamanager = DataManager(this)

        refresh(datamanager)
        swipeRefresh.setOnRefreshListener { refresh(datamanager)
        }

    }

    private fun displayUpdatedContent() {
        myAdapter.contactList.clear()
        myAdapter.contactList.addAll(databasemanager.readContactList()!!.toList())
        myAdapter.notifyDataSetChanged()
    }

    private fun refresh(datamanager: DataManager) {
        datamanager.refresh()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : SingleObserver<List<ContactService.ApiContact>> {
            override fun onSuccess(contactList: List<ContactService.ApiContact>) {
                swipeRefresh.isRefreshing = false
                displayUpdatedContent()
            }

            override fun onSubscribe(d: Disposable?) {
            }

            override fun onError(e: Throwable?) {
                swipeRefresh.isRefreshing = false
                Log.i("DataManager", "Failed to subscribe")
            }
        })
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {

            R.id.action_add -> {
                val intent = Intent(this, AjoutContact::class.java)
                startActivityForResult(intent, 1)

                return true
            }
            R.id.Delete -> {
                val intent1 = Intent(this, DeleteActivity::class.java)
                startActivityForResult(intent1, 2)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            1,2 -> {
                if (resultCode == Activity.RESULT_OK) {
                    displayUpdatedContent()
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        refresh(datamanager)

    }
}
